keywords/MobileRestClient.py

```diff
@@ -1131,24 +1130,19 @@ def add_docs(self, url, db, number, id_prefix, auth=None, channels=None, generat
             if channels is not None:
                 doc_body["channels"] = channels
 
-            data = json.dumps(doc_body)
+            if attachments_generator:
+                types.verify_is_callable(attachments_generator)
+                attachments = attachments_generator()
+                doc_body["_attachments"] = {att.name: {"data": att.data} for att in attachments}
 
             if id_prefix is None:
                 doc_id = str(uuid.uuid4())
             else:
                 doc_id = "{}_{}".format(id_prefix, i)
 
-            if auth_type == AuthType.session:
-                resp = self._session.put("{}/{}/{}".format(url, db, doc_id), data=data, cookies=dict(SyncGatewaySession=auth[1]))
-            elif auth_type == AuthType.http_basic:
-                resp = self._session.put("{}/{}/{}".format(url, db, doc_id), data=data, auth=auth)
-            else:
-                resp = self._session.put("{}/{}/{}".format(url, db, doc_id), data=data)
-
-            log_r(resp, info=False)
-            resp.raise_for_status()
+            doc_body["_id"] = doc_id
 
-            doc_obj = resp.json()
+            doc_obj = self.add_doc(url, db, doc_body, auth=auth)
```

Comment:
add_doc uses post by default, so the behavior is different now. `use_post=False` will give the same behavior